src/editortoolkit.cpp

```diff
@@ -924,9 +932,101 @@ bool EditorToolkit::Ungroup(std::string groupType, std::vector<std::string> elem
+{
+    m_editInfo = "";
+    //Check if you can get drawing page
+    if(!m_doc->GetDrawingPage()) {
+        LogError("Could not get the drawing page.");
+        return false;
+    }
+    Neume *el = dynamic_cast<Neume *> (m_doc->GetDrawingPage()->FindChildByUuid(elementId));
+    Nc *firstChild, *prevNc;
+
+    //Get children of neume. Keep the first child and delete the others.
+    AttComparison ac(NC); 
+    ArrayOfObjects children;
+    el->FindAllChildByComparison(&children, &ac);
+    for (auto it = children.begin(); it != children.end(); ++it) {
+        if(children.begin() == it){
+            firstChild = dynamic_cast<Nc *> (*it);     
+        }
+        else{
+            el->DeleteChild(*it);
+        }
+    }
+
+    //Get the coordinates of the remaining child.
+    int initialUlx = firstChild->GetZone()->GetUlx();
+    int initialUly = firstChild->GetZone()->GetUly();
+    int initialLrx = firstChild->GetZone()->GetLrx();
+    int initialLry = firstChild->GetZone()->GetLry();
+
+    Staff *staff = dynamic_cast<Staff *> (el->GetFirstParent(STAFF));
+    assert(staff);
+    Facsimile *facsimile = m_doc->GetFacsimile();
+
+    const int noteHeight = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 2);
+    const int noteWidth = (int)(m_doc->GetDrawingDoubleUnit(staff->m_drawingStaffSize) / 1.4);
+    prevNc = firstChild;
+
+    //Iterate throught the contour and build the new grouping.
+    for(auto it = contour.begin(); it != contour.end(); ++it) {
+        Nc *newNc = new Nc();
+        Zone *zone = new Zone();
+        int newUlx = initialUlx + noteWidth;
+        int newLrx = initialLrx + noteWidth;
+        int newUly, newLry;
+
+        newNc->SetPname(prevNc->GetPname());
+        newNc->SetOct(prevNc->GetOct());  
+
+        if((*it) == 'u'){
+            newUly = initialUly - noteHeight;
+            newLry = initialLry - noteHeight;
+            newNc->AdjustPitchByOffset(1);
+        }
+        else if((*it) == 'd'){
+            newUly = initialUly + noteHeight;
+            newLry = initialLry + noteHeight;
+            newNc->AdjustPitchByOffset(-1);
+        }
+        else if((*it) == 's'){
+            newUly = initialUly;
+            newLry = initialLry;
+        }
+        else{
+            LogMessage("Unsupported character in contour.");
+            return false;
+        }
+        zone->SetUlx(newUlx);
+        zone->SetUly(newUly);;
+        zone->SetLrx(newLrx);
+        zone->SetLry(newLry);
+
+        newNc->SetZone(zone);
+        newNc->SetFacs(zone->GetUuid());
+
+        Surface *surface = dynamic_cast<Surface *>(facsimile->FindChildByType(SURFACE));
+        assert(surface);
+        surface->AddChild(zone);
+
+        el->AddChild(newNc);
+
+        initialUlx = newUlx;
+        initialUly = newUly;
+        initialLrx = newLrx;
+        initialLry = newLry;
+        prevNc = newNc;
+
+    }
+    m_editInfo = firstChild->GetUuid();
```

Comment:
The UUID should probably be that of the neume, not the first neume component. I don't see how that would be as useful.

Reply:
fixed in fbe8061